packages/contracts-bedrock/contracts/L1/winddown/BalanceClaimer.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Libraries
+import { MerkleProof } from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import { Initializable } from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+
+// Interfaces
+import { IEthBalanceWithdrawer } from "../interfaces/winddown/IEthBalanceWithdrawer.sol";
+import { IErc20BalanceWithdrawer } from "../interfaces/winddown/IErc20BalanceWithdrawer.sol";
+import { IBalanceClaimer } from "../interfaces/winddown/IBalanceClaimer.sol";
+import { Semver } from "../../universal/Semver.sol";
+
+/**
+  * @custom:proxied
+  * @notice Contract that allows users to claim and withdraw their eth and erc20 balances
+ */
+contract BalanceClaimer is Initializable, Semver, IBalanceClaimer {
+    /// @notice the root of the merkle tree
+    bytes32 public root;
```

Comment:
I noticed this is not immutable, but is only written to in the constructor. Making it immutable could save some gas, and for testing we could etch a new contract on the address of the current one

Reply:
++

Reply:
++